packages/liferay-npm-scripts/src/utils/soy.js

```diff
@@ -4,35 +4,42 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-const fs = require('fs');
+const fs = require('fs-extra');
 const path = require('path');
-const rimraf = require('rimraf');
 
 const expandGlobs = require('./expandGlobs');
 const generateSoyDependencies = require('./generateSoyDependencies');
 const getMergedConfig = require('./getMergedConfig');
 const log = require('./log');
+const runBabel = require('./runBabel');
 const spawnSync = require('./spawnSync');
 
 const BUILD_CONFIG = getMergedConfig('npmscripts', 'build');
+const SOY_TEMP_DIR = path.join(BUILD_CONFIG.temp, 'soy');
 
 /**
  * Compiles soy files by running `metalsoy` bin with specified soy dependencies
  */
 function buildSoy() {
 	spawnSync('metalsoy', [
+		'-s',
+		path.join(BUILD_CONFIG.input, '**', '*.soy'),
+		'-d',
+		SOY_TEMP_DIR,
 		'--soyDeps',
 		generateSoyDependencies(BUILD_CONFIG.dependencies),
 		'--externalMsgFormat',
 		"Liferay.Language.get('$2')"
 	]);
+
+	runBabel(SOY_TEMP_DIR, '--out-dir', BUILD_CONFIG.output, '--source-maps');
```

Comment:
Does Babel do anything useful for us here? If I understand correctly, before this PR, we used to run:

1. Compile soy (in place)
2. Babel
3. Bundler

So, we were using Babel on the compiled soy before, but I don't know what is in there that would need transforming with Babel (the bit we do need — wrapping in `Liferay.Loader.define` etc — happens in step 3 anyway).

Reply:
The soy compiler generates `import from...` statements, that's why it needs transpiling. :woman_shrugging:

Reply:
😱